### PR TITLE
#411: Drop unused dependencies + Separate devDependencies

### PR DESCRIPTION
**Pull request by fregante**
Replaces https://github.com/pixiebrix/pixiebrix-extension/pull/462

Part of https://github.com/pixiebrix/pixiebrix-extension/issues/411

### CRX file

16.4MB -> 14.6MB (-11%)

### Installed dependencies 

1694 -> 1412 dependencies (-16% fewer things for `npm` to install)

### CODE REVIEW

File: src/auth/token.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@
  */
 
 import { readStorage, setStorage } from "@/chrome";
-import equal from "deep-equal";
+import equal from "fast-deep-equal";
```
Comment (fregante) [inline src/auth/token.ts:19]: Why?

- deep-equal dependency graph: https://npmgraph.js.org/?q=deep-equal
- fast-deep-equal dependency graph: https://npmgraph.js.org/?q=fast-deep-equal

Both packages are downloaded in the millions/week

File: src/utils.ts [hunk 2]
```diff
@@ -240,7 +240,7 @@ export function getPropByPath(
     proxy = noopProxy,
   }: { args?: object; proxy?: ReadProxy } | undefined = {}
 ): unknown {
-  // consider using jsonpath syntax https://www.npmjs.com/package/jsonpath
+  // consider using jsonpath syntax https://www.npmjs.com/package/jsonpath-plus
```
Comment (fregante) [inline src/utils.ts:243]: It appears that the repo uses `jsonpath-plus` everywhere, even though it installed `jsonpath`

**Review by fregante — COMMENTED** [review]
You can review the two commits separately to see what was removed and what was simply moved

**Review by twschiller — APPROVED** [review]

**Comment by fregante** [comment]
In [`ab53d07`](https://github.com/pixiebrix/pixiebrix-extension/pull/467/commits/ab53d077c065ec9f2ced5812585cf7ea84aad3ce) it appears that some installs were mistakes (e.g. `$ npm install npm install`) but others were probably along the lines of "I want to use this, but no time right now," so you might want to take a look if there's anything _you'd like_ to restore at some point.

**Pull request merged**